pkg/controller/performanceprofile/performanceprofile_controller.go

```diff
@@ -103,54 +210,350 @@ func (r *ReconcilePerformanceProfile) Reconcile(request reconcile.Request) (reco
+	if mcpMutated == nil &&
+		mcMutated == nil &&
+		kcMutated == nil &&
+		fgMutated == nil &&
+		networkLatencyTunedMutated == nil &&
+		realTimeKernelTunedMutated == nil {
+		// if nothing needed to be updated and machine config pool paused, unpause it
+		mcpUpdate, err := r.getMachineConfigPool(mcp.Name)
+		if err != nil {
+			return nil, err
+		}
+
+		updated, err := r.pauseMachineConfigPool(mcpUpdate, false)
+		if err != nil {
+			return nil, err
+		}
+
+		if updated {
+			return &reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+		return nil, nil
+	}
+
+	// create machine config pool and pause it
+	if mcpMutated != nil && created {
+		if err := r.createOrUpdateMachineConfigPool(mcpMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	mcpUpdate, err := r.getMachineConfigPool(mcp.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	updated, err := r.pauseMachineConfigPool(mcpUpdate, true)
+	if err != nil {
+		return nil, err
+	}
+
+	// we want to give time to the machine-config controllers to get updated values
+	if updated {
+		return &reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+	}
+
+	// update the machine config pool after that we paused it
+	if mcpMutated != nil && !created {
+		if err := r.createOrUpdateMachineConfigPool(mcpMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	if mcMutated != nil {
+		if err := r.createOrUpdateMachineConfig(mcMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	if networkLatencyTunedMutated != nil {
+		if err := r.createOrUpdateTuned(networkLatencyTunedMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	if realTimeKernelTunedMutated != nil {
+		if err := r.createOrUpdateTuned(realTimeKernelTunedMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	if fgMutated != nil {
+		if err := r.createOrUpdateFeatureGate(fgMutated); err != nil {
+			return nil, err
+		}
+
+		// feature gate resource should be updated before KubeletConfig creation, otherwise
+		// we will lack TopologyManager feature gate under the kubelet configuration
+		// see - https://bugzilla.redhat.com/show_bug.cgi?id=1788061#c3
+		// we want to give time to the kubelet-config controllers to get updated feature gate resource
+		return &reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+	}
+
+	if kcMutated != nil {
+		if err := r.createOrUpdateKubeletConfig(kcMutated); err != nil {
+			return nil, err
+		}
+	}
+
+	return &reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+}
+
+func (r *ReconcilePerformanceProfile) pauseMachineConfigPool(mcp *mcov1.MachineConfigPool, pause bool) (bool, error) {
+	if mcp.Spec.Paused == pause {
+		return false, nil
+	}
+
+	mcp.Spec.Paused = pause
+	return true, r.client.Update(context.TODO(), mcp)
+}
+
+func (r *ReconcilePerformanceProfile) validatePerformanceProfileParameters(performanceProfile *performancev1alpha1.PerformanceProfile) error {
```

Comment:
I added all parameters that needed for scripts as required, otherwise, we can have some unpredicted behavior under our scripts. To change it we will need to change our scripts, so I prefer to leave it for another PR, opened https://github.com/openshift-kni/performance-addon-operators/issues/33

Reply:
alrighty, as long as we're tracking this, thanks